presto-main/src/main/java/com/facebook/presto/type/DecimalParametricType.java

```diff
@@ -37,26 +36,21 @@ public String getName()
     }
 
     @Override
+    @NotNull
     public Type createType(List<TypeParameter> parameters)
     {
+        checkArgument(parameters.size() <= 2, "Expected no more than 2 literal parameters for DECIMAL");
+
         if (parameters.size() == 0) {
-            // we need support for that for support of unparametrized TypeSignatures which are
-            // used for defining DECIMAL related functions. See e.g. DecimalOperators class.
-            return ShortDecimalType.createUnparametrizedDecimal();
+            return ShortDecimalType.createDecimalType();
```

Comment:
Call static `createDecimalType` on `DecimalType` class. Not on `ShortDecimalType` sub-class.